middleware/rewrite/rewrite.go

```diff
@@ -119,32 +109,19 @@ func (r *RegexpRule) Rewrite(req *http.Request) bool {
 		return false
 	}
 
+	// include trailing slash in regexp if present
+	start := len(r.Base)
+	if strings.HasSuffix(r.Base, "/") {
+		start -= 1
+	}
+
 	// validate regexp
-	if !r.MatchString(rPath[len(r.Base):]) {
+	if !r.MatchString(rPath[start:]) {
 		return false
 	}
 
-	to := r.To
-
-	// check variables
-	for _, v := range regexpVars {
-		if strings.Contains(r.To, v) {
-			switch v {
-			case "{path}":
-				to = strings.Replace(to, v, req.URL.Path[1:], -1)
-			case "{query}":
-				to = strings.Replace(to, v, req.URL.RawQuery, -1)
-			case "{frag}":
-				to = strings.Replace(to, v, req.URL.Fragment, -1)
-			case "{file}":
-				_, file := path.Split(req.URL.Path)
-				to = strings.Replace(to, v, file, -1)
-			case "{dir}":
-				dir, _ := path.Split(req.URL.Path)
-				to = path.Clean(strings.Replace(to, v, dir, -1))
-			}
-		}
-	}
+	// replace variables
+	to := path.Clean(middleware.NewReplacer(req, nil, "").Replace(r.To))
```

Comment:
Good call on path.Clean! I might need to go add that in other parts of Caddy that use the replacer on paths...